libs/blocks/quiz/utils.js

```diff
@@ -202,18 +202,20 @@ export const nestedFragments = (
 
 const getNestedFragments = (resultResources, productCodes, fragKey) => {
   const fragArray = [];
-  resultResources?.data?.forEach((row) => {
-    if (productCodes.length > 0 && productCodes.includes(row.product)) {
-      insertFragment();
-    }
+  productCodes?.forEach((product) => {
+    resultResources?.data?.forEach((row) => {
+      if (product && product === row?.product) {
+        insertFragment();
+      }
 
-    function insertFragment() {
-      if (row[fragKey]) {
-        row[fragKey].split(',').forEach((val) => {
-          fragArray.push(val.trim());
-        });
+      function insertFragment() {
+        if (row[fragKey]) {
+          row[fragKey]?.split(',').forEach((val) => {
+            fragArray.push(val.trim());
+          });
+        }
       }
-    }
+    });
```

Comment:
can we move this function declaration above line 207 `if (product && product === row?.product) {`

Reply:
Hi @sabyamon, 

As far as I can see, when I set a breakpoint in storeResultInLocalStorage(), I believe the fragments are coming in the correct order. For some reason the results page doesn't load from the uar-nested-frag-in-order branch, but I'm approving based upon what I see in that last method call before it redirects. 

https://uar-nested-frag-in-order--milo--adobecom.hlx.live/drafts/xiasun/uar-results-block/uar-results?primary=lr-edu,pr-edu&quizKey=cc-quiz-en-US

If you can work on the comments from @JasonHowellSlavin, I think we should be good to go afterwards.

Reply:
hi @JasonHowellSlavin, I am a bit unclear about why you want me to move the `insertFragment` function above line 207. Can you please explain?

Reply:
Sure. `insertFragment` is used on 208, but defined on 211. If running eslint, this should set off an error: `no-use-before-defined`. Though the function will be hoisted, it is preferable to not rely on hoisting.